code/integration-tests/runtime-tests/test/tests/xcmp/transfersTestHelper.ts

```diff
@@ -0,0 +1,677 @@
+  return new BN(free);
+}
+
+export function calculateTransferredAmount(transferredAmount: number, chainXTransferFee: number) {
+  return new BN(transferredAmount * 10 ** 12).sub(new BN(chainXTransferFee));
+}
+
+export async function changeGasToken(api: ApiPromise, wallet: KeyringPair, assetId: number) {
+  const payer = wallet.publicKey;
+  const asset = api.createType("Option<u128>", assetId);
+  return await sendAndWaitForSuccess(
+    api,
+    wallet,
+    api.events.tokens.Reserved.is,
+    api.tx.assetTxPayment.setPaymentAsset(payer, asset)
+  );
+}
+
+export async function sendPicaToAnotherAccount(
+  api: ApiPromise,
+  senderWallet: KeyringPair,
+  receiverWallet: KeyringPair,
+  assetdId: number,
+  amount: number
+) {
+  const asset = api.createType("u128", 1);
+  const dest = api.createType("MultiAddress", {
+    Id: api.createType("AccountId", receiverWallet.publicKey)
+  });
+  const amountP = api.createType("u128", Pica(amount));
+  const keepAlive = api.createType("bool", "No");
+  return await sendAndWaitForSuccess(
+    api,
+    senderWallet,
+    api.events.balances.Transfer.is,
+    api.tx.assets.transfer(asset, dest, amountP, keepAlive)
+  );
+}
+
+export async function fetchTotalIssuance(api: ApiPromise, assetId: number) {
+  const balance = await api.query.tokens.totalIssuance(api.createType("u128", assetId));
+  return new BN(balance.toString());
+}
+
+export async function fetchXChainTokenBalances(
+  apis: ApiPromise[],
+  wallets: (string | KeyringPair)[],
+  assetId: number[]
+) {
+  const promises = apis
+    .map((api, index) => {
+      return wallets.map(wallet => {
+        if (assetId[index] === 0) {
+          return fetchNativeBalance(api, wallet);
+        } else if (assetId[index] === 1984 || assetId[index] === 1985) {
+          return fetchTokenBalanceOnStatemine(api, wallet, assetId[index]);
+        }
+        return fetchTokenBalance(api, wallet, assetId[index]);
+      });
+    })
+    .flat();
+  return await Promise.all(promises);
+}
+
+export async function sendAssetToRelaychain(
+  api: ApiPromise,
+  senderWallet: KeyringPair,
+  receiveeWallet: KeyringPair,
+  amount: number,
+  destinationWeight: number
+) {
+  const currencyId = api.createType("u128", 4);
+  const amountP = api.createType("u128", Pica(amount));
+  const dest = setBeneficiary(api, receiveeWallet, 1);
+  await sendAndWaitForSuccess(
+    api,
+    senderWallet,
+    api.events.xTokens.TransferredMultiAssets.is,
+    api.tx.xTokens.transfer(currencyId, amountP, dest, destinationWeight)
+  );
+}
+
+export async function fetchNativeBalance(api: ApiPromise, wallet: KeyringPair | string) {
+  if (typeof wallet === "string") {
+    const accountId = await api.createType("AccountId32", wallet);
+    const balanceRaw = await api.query.system.account(accountId);
+    return new BN(balanceRaw.data.free.toString());
+  }
+  const accountId = await api.createType("AccountId32", wallet.publicKey);
```

Comment:
(typescript:S4123): `"await" should only be used with promises`

`api.createType(...)` is not an async method.